### ISSUE TITLE
compile openssl (version >= 1.1.0) for Universal Windows Platform ARM architecture

### ISSUE THREAD

**Issue by neerajgoel** [openssl/openssl#4156]
I am able to build openssl 1.1.0v for win64 and win32. But facing problems to build for ARM architecure. For **Win32**, supported architecture is **VC-WIN32** and for **Win64** supported architecture is **VC-WIN64**.
**For ARM, there is no such option in supported architecture list**. 
Please help asap.

**Comment by richsalz** [comment]
Do you mean 1.0.1?  1.1.0v is not a release.
If you mean 1.0.1, you have to upgrade; that release is not supported.

**Comment by michaelsaxton** [comment]
I too looked for that but you just use the VC-Win32 option from your ARM Cross Tools Command Prompt 

perl <src folder>\openssl-1.1.0e\Configure VC-WIN32 no-shared no-asm -D_ARM_WINAPI_PARTITION_DESKTOP_SDK_AVAILABLE

You may have to adjust your LIB path - at least I did.

**Comment by GyrosGeier** [comment]
This gives a missing symbol `OPENSSL_UplinkTable`.

[Full build log](http://darine.hogyros.de:8080/job/windows-openssl-msvc/build=debug,cpu=arm,label=msvc/92/consoleFull

**Comment by levitte** [comment]
@GyrosGeier, does this patch help?

```
diff --git a/Configurations/10-main.conf b/Configurations/10-main.conf
index 39b89e7e98..da0bc0edc0 100644
--- a/Configurations/10-main.conf
+++ b/Configurations/10-main.conf
@@ -1376,7 +1376,7 @@ sub vms_info {
         # x86 Win32 target defaults to ANSI API, if you want UNICODE,
         # configure with 'perl Configure VC-WIN32 -DUNICODE -D_UNICODE'
         inherit_from     => [ "VC-noCE-common", asm("x86_asm"),
-                              sub { $disabled{shared} ? () : "uplink_common" } ],
+                              sub { $disabled{shared} ? () : "x86_uplink" } ],
         as               => sub { vc_win32_info()->{as} },
         asflags          => sub { vc_win32_info()->{asflags} },
         asoutflag        => sub { vc_win32_info()->{asoutflag} },
```

@dot-asm, can you remind me why `x86_uplink` isn't used in `VC-WIN32`, while `ia64_uplink` and `x86_64_uplink` are used, in `VC-WIN64I` and `VC-WIN64A`?

**Comment by levitte** [comment]
Well, I'm honestly doubtful, since ARM is a different processor architecture, so using `VC-WIN32` (which is for x86) seems wrong for starters....  I didn't think if that a moment ago, and quite frankly, this line kind of threw me off:

```
set vc_cpu=amd64_arm 
```

Looks like a mixup, doesn't it?

**Comment by GyrosGeier** [comment]
No, that is the config for amd64 host, arm target.

**Comment by levitte** [comment]
Ah, cross compiling?  I'm sorry, that's way over my head on Windows :-/

**Comment by GyrosGeier** [comment]
Well, unless you try to run one of the binaries, it is transparent. Normal cl.exe just outputs ARM binaries.

If you do try to run a binary, you get rundll32.exe displaying a modal dialog even in noninteractive sessions where there is no one to close it.

**Comment by levitte** [comment]
Ok, I get that.  But still, you will then need a `ms\uplink-arm{something}.pl`, or no uplink stuff at all.  That'll be a new config target either way...

One way you can try this out is by copying the `VC-WIN64A` stuff from `Configurations\10-main.conf` into another target (`VC-WIN64A-ARM`?), but starting like this:

```
    "VC-WIN64A-ARM" => {
        inherit_from     => [ "VC-WIN64-common" ],
        as               => sub { vc_win64a_info()->{as} },
```

NOTE: *experimental, and you're the guinea pig*

**Comment by GyrosGeier** [comment]
Okay, this doesn't seem to be entirely wrong.

https://github.com/GyrosGeier/openssl/tree/vc-arm

has a minimal implementation. I don't reference this Issue, because the branch doesn't address UWP (that's why I need the additional preprocessor symbols for).

It survived a [test build](http://darine.hogyros.de:8080/view/all/job/windows-openssl-msvc/97/), but I haven't run tests, primarily because I don't have hardware.

**Comment by levitte** [comment]
Pertty cool

**Comment by dot-asm** [comment]
Resolved in #5502.

**Comment by miguelcruztitla12-coder** [comment]
> I am able to build openssl 1.1.0v for win64 and win32. But facing problems to build for ARM architecure. For **Win32**, supported architecture is **VC-WIN32** and for **Win64** supported architecture is **VC-WIN64**. **For ARM, there is no such option in supported architecture list**. Please help asap.

https://github.com/openssl/openssl/issues/4156#issue-249955668
Leobardocruzticla@gmail.com
